subprojects/docs/src/docs/userguide/build_cache.adoc

```diff
@@ -187,132 +187,153 @@ This means that tasks with different task paths can re-use each other's outputs
-    <sourcefile file="settings.gradle" snippet="configure-http-build-cache"/>
-</sample>
-++++
+=== Example: Configure remote HTTP cache
+
+[source.multi-language-sample,groovy]
+.settings.gradle
+----
+include::{samplesPath}/buildCache/configure-built-in-caches/settings.gradle[tag=configure-http-build-cache]
+----
 
 [NOTE]
 ====
 You may encounter problems with an untrusted SSL certificate when you try to use a build cache backend with an HTTPS URL.
 The ideal solution is for someone to add a valid SSL certificate to the build cache backend, but we recognize that you may not be able to do that.
-In that case, set api:org.gradle.caching.http.HttpBuildCache#isAllowUntrustedServer()[] to `true`:
+In that case, set link:{groovyDslPath}/org.gradle.caching.http.HttpBuildCache.html#org.gradle.caching.http.HttpBuildCache:allowUntrustedServer[HttpBuildCache.isAllowUntrustedServer()] to `true`:
 
-++++
-<sample id="httpBuildCacheAllowUntrustedServer" dir="buildCache/http-build-cache" title="Allow untrusted SSL certificate for HttpBuildCache">
-    <sourcefile file="settings.gradle" snippet="allow-untrusted-server"/>
-</sample>
-++++
+[source.multi-language-sample,groovy]
+.settings.gradle
+----
+include::{samplesPath}/buildCache/http-build-cache/settings.gradle[tag=allow-untrusted-server]
+----
 
 This is a convenient workaround, but you shouldn’t use it as a long-term solution.
 ====
 
 [[sec:build_cache_configure_use_cases]]
-==== Configuration use cases
+=== Configuration use cases
 
 The recommended use case for the build cache is that your continuous integration server populates the remote build cache from clean builds while developers load from the remote build cache and store in the local build cache.
 The configuration would then look as follows.
 
-++++
-<sample id="developerCiSetup" dir="buildCache/developer-ci-setup" title="Recommended setup for CI push use case">
-    <sourcefile file="settings.gradle" snippet="developer-ci-setup"/>
-</sample>
-++++
+=== Example: Recommended setup for CI push use case
+
+[source.multi-language-sample,groovy]
+.settings.gradle
+----
+include::{samplesPath}/buildCache/developer-ci-setup/settings.gradle[tag=developer-ci-setup]
+----
 
 If you use a `buildSrc` directory, you should make sure that it uses the same build cache configuration as the main build.
 This can be achieved by applying the same script to `buildSrc/settings.gradle` and `settings.gradle` as shown in the following example.
 
-++++
-<sample id="buildCacheBuildSrc" dir="buildCache/build-src" title="Consistent setup for buildSrc and main build">
-    <sourcefile file="settings.gradle" snippet="configure-build-src-build-cache"/>
-    <sourcefile file="buildSrc/settings.gradle" snippet="configure-build-src-build-cache"/>
-    <sourcefile file="gradle/buildCacheSettings.gradle" snippet="configure-build-src-build-cache"/>
-</sample>
-++++
+=== Example: Consistent setup for buildSrc and main build
```

Comment:
❌ Something went wrong the the file titles here:

![image](https://user-images.githubusercontent.com/423186/43023883-af1e0bf8-8c6c-11e8-8fff-675d54dabe83.png)

Reply:
Fixed

Reply:
Now the syntax highlighting seems to be gone.
![image](https://user-images.githubusercontent.com/423186/43142677-3c2cd2ba-8f59-11e8-9d69-39d48958b0ba.png)

Reply:
It looks good when building locally 👍